extensions/battleroyale/standalone.mjs

```diff
@@ -0,0 +1,542 @@
+   */
+  async s2s_sf_quitBattleroyale(srcExt, playerID, kwargs, sfInfo) {
+    const {next} = kwargs;
+    if (this.participatePlayerIDs.has(playerID)) {
+      this.participatePlayerIDs.delete(playerID);
+    }
+    return next;
+  }
+
+  /**
+   * provide state function to npc
+   * @param {*} srcExt
+   * @param {*} registerFunc
+   */
+  async s2s_provideStateFunc(srcExt, registerFunc) {
+    this.helper.callS2sAPI(srcExt, registerFunc, this.helper.getListOfStateFunctions(this));
+  }
+
+  /**
+   * kill player and tp
+   * @param {String} playerID player ID
+   * @param {Boolean} endGame kill player and terminate
+   */
+  async killPlayer(playerID, endGame = false) {
+    let target;
+    if (this.playerOldPosition.has(playerID)) {
+      target = this.playerOldPosition.get(playerID);
+    } else { // default
+      const player = this.helper.gameState.getPlayer(playerID);
+      target = player.mapCoord.copy();
+      target.x = 1;
+      target.y = 1;
+    }
+    await this.helper.teleport(playerID, target, true);
+    this.playerOldPosition.delete(playerID);
+    this.participatePlayerIDs.delete(playerID);
+    if (this.participatePlayerIDs.size <= 1 && !endGame) {
+      await this.terminateGame();
+    }
+  }
+
+  /**
+   * @param {String} playerID playerID
+   */
+  async teleportPlayer(playerID) {
+    const player = this.helper.gameState.getPlayer(playerID);
+    const target = player.mapCoord.copy();
+    target.x = this.arenaPos.x;
+    target.y = this.arenaPos.y;
+    await this.helper.teleport(playerID, target, true);
+  }
+
+  /**
+   * game is over give trophy and quit
+   */
+  async terminateGame() {
+    for (const playerID of this.participatePlayerIDs) {
+      await this.helper.callS2cAPI(playerID, 'notification', 'showNotification', 3000, '[battleroyale] You Win!');
+      // TODO: give item
+    }
+
+    for (const playerID of this.participatePlayerIDs) {
+      await this.killPlayer(playerID, true);
+    }
+
+    await this.resetGame();
+  }
+
+  /**
+   * reset the whole game
+   */
+  async resetGame() {
+    this.gameStarted = false;
+    this.fires = new Set();
+    this.cooldown = new Set();
+    this.fireCounter = 0;
+    this.gameStarted = false;
+    for (const mapName of this.arenaOfMaps.keys()) {
+      await this.helper.broadcastCellSetUpdateToAllUser(
+          'update',
+          mapName,
+          CellSet.fromObject({
+            name: LAYER_FIRE.layerName,
+            cells: Array.from(this.fires),
+          }),
+      );
+      await this.helper.broadcastCellSetUpdateToAllUser(
+          'update',
+          mapName,
+          CellSet.fromObject({
+            name: LAYER_BULLET.layerName,
+            cells: Array.from(
+                this.bullets,
+                ([_, v]) => v['mapCoord'],
+            ),
+          }),
+      );
+    }
+    for (const ID of this.updateMapIntervalIDs) {
+      clearInterval(ID);
+    }
+    this.updateMapIntervalIDs = [];
```

Comment:
Just a kind reminder that if `this.updateMapIntervalIDs` is used in other places, you should consider clearing the array instead of assigning a new array since you might want to keep the object referencing the same object.